CMakeLists.txt

```diff
@@ -24,7 +24,13 @@ else ()
     message(FATAL_ERROR "Compiler ${CMAKE_CXX_COMPILER} has no C++11 support")
 endif ()
 
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DBOOST_SYSTEM_NO_DEPRECATED -fopenmp -Wall -Wextra")
+if (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+    set(MACOSX TRUE)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DBOOST_SYSTEM_NO_DEPRECATED -DNO_OPENMP -Wall -Wextra")
+else ()
+    find_package(OPENMP 4.5 REQUIRED)
```

Comment:
I am wondering if we should downgrade the OpenMP version required from 4.5 to 4.0. The reason is that we require C++11 support, and gcc fully supports C++11 from gcc 4.8.1 (https://gcc.gnu.org/projects/cxx-status.html#cxx11), but OpenMP 4.5 is only supported in gcc 6.1 (https://www.openmp.org/resources/openmp-compilers-tools/). Downgrading to OpenMP 4.0 means that the user needs only gcc 4.9.0 to compile. Although I did not test if it works with OpenMP 4.0, I am guessing so since there are no features exclusively from OpenMP 4.5 that we use (see https://www.openmp.org/wp-content/uploads/openmp-4.5.pdf pages 348-350).

Reply:
I've downgraded to 4.0. I can't test if it compiles fine on my Mac, so if it doesn't we can change the number again later.